woocommerce/woocommerce-android
# Redesign products section when it's nonempty

Closes: #9947 


### Description

The PR introduces enhancements to the products section, specifically when it contains entries. A notable addition is the "+" button situated in the products header, which facilitates the addition of new products.

Design: https://www.figma.com/file/ichqhxDjbmZZmvJaJkGhkj/Simple-Payments-%26-Order-Creation?type=design&node-id=165-46199&mode=design&t=ddOsdnzmq3KlU1e8-0

### Testing instructions

1. Navigate to the `Orders` tab
2. Click on the "+" icon to navigate to the order creation flow
3. Add a product
4. Notice the new products section UI should be as per design.

### Images/gif


<img src="https://github.com/woocommerce/woocommerce-android/assets/1331230/4af05b7a-841b-484e-85b2-cd39a5a3ee29" alt="new_products_nonempty_section" width = 360px>
